drivers/hwinfo/hwinfo_andes.c

```diff
@@ -113,3 +125,16 @@ int z_impl_hwinfo_get_supported_reset_cause(uint32_t *supported)
 
 	return 0;
 }
+
+static int hwinfo_andes_init(void)
```

Comment:
Why does the driver has to be ready at init if you check the readiness in each call?

Reply:
Because this can help us find the errors even when not using function calls.
If we only check at the init and CONFIG_LOG isn't set, we cannot find the errors exactly when calling the function.

Reply:
@alexanderwachter Have I addressed your questions adequately, or do you have any additional inquiries or suggestions?

Reply:
Sorry, forgot this PR.
In my opinion we should not have an init function that only checks if another driver failed, because there is no particular device behind hwinfo.
The device ready for the underlaying device is checked in each call anyway.

Reply:
Thanks for the review. I've edited the changes.